src/main/java/org/opennars/main/Nar.java

```diff
@@ -193,19 +184,24 @@ public boolean isEnabled() {
     private boolean stopped = false;
     private boolean threadYield;
 
-    public Nar() {
+    public Nar(long narId) {
         final Plugins b = new Plugins();
         final Memory m = new Memory(new RuntimeParameters(),
                 new LevelBag(narParameters.CONCEPT_BAG_LEVELS, narParameters.CONCEPT_BAG_SIZE),
                 new LevelBag<>(Parameters.NOVEL_TASK_BAG_LEVELS, Parameters.NOVEL_TASK_BAG_SIZE),
                 new LevelBag<>(Parameters.SEQUENCE_BAG_LEVELS, Parameters.SEQUENCE_BAG_SIZE),
                 new LevelBag<>(Parameters.OPERATION_BAG_LEVELS, Parameters.OPERATION_BAG_SIZE));
         this.memory = m;
+        this.memory.narId = narId;
         this.param = m.param;
         for (final Operator o : Operators.get(this))
             this.memory.addOperator(o);
         new Plugins().init(this);
     }
+    
+    public Nar() {
+        this(java.util.UUID.randomUUID().getLeastSignificantBits());
```

Comment:
This still get's the mac and some other pieces - but it's fine for this PR

Reply:
I agree, we can improve this.